chia/wallet/trading/trade_store.py

```diff
@@ -140,6 +142,17 @@ async def create(
             except aiosqlite.OperationalError:
                 pass  # ignore what is likely Duplicate column error
 
+            try:
+                await conn.execute("CREATE TABLE trade_record_times(trade_id blob PRIMARY KEY, valid_times blob)")
```

Comment:
I prefer not to create a new table just for this column. It is also easier when you read the data, you don't have to read data from two tables and ensure they are synced. You can take a look of how we add new column in other tables like coin_store.

Reply:
You can just add a line after ln140
await conn.execute("ALTER TABLE trade_records ADD COLUMN valid_times blob DEFAULT EMPTY_BLOB()")

To make this backward compatible, the key is the Alter SQL must be the last Alter sentences. This means you are not inserting a column between existing columns. So when the user rollback to the old version, the index of row[] will not change.

Reply:
That was my instinct too but @emlowe raised an issue where when you have statements like this: https://github.com/Chia-Network/chia-blockchain/blob/e7778c77496fdb8ab822ef60f2722c44510245de/chia/wallet/wallet_transaction_store.py#L94 then altering the table actually is bad for forwards compatibility because then sqlite will error with something like `table transaction_records got N+1 values, was expecting N`